include/service/entities/site_service.js

```diff
@@ -289,6 +289,9 @@ module.exports = function SiteServiceModule(pb) {
      */
     SiteService.prototype.editSite = function(options, cb) {
         cb = cb || util.cb;
+        if (!options.prevHostnames) {
+            options.prevHostnames = [];
+        }
```

Comment:
Each service that inherits from BOS should register for and handle some events.  The majority of the time this should be "merge", "format", and "validate".  

You are correct, ensuring that the prevHostnames property is an array should not be performed in the format method.  Instead, it should be in the merge since it is a protected property that is essentially, read-only.  I would do the check in the merge handler.  My apologies.